ORBIT/phases/design/electrical_export.py

```diff
@@ -327,63 +378,87 @@ def substation_cost(self):
             + self.switchgear_cost
             + self.converter_cost
             + self.dc_breaker_cost
-            + self.ancillary_system_cost
+            + self.ancillary_system_costs
             + self.land_assembly_cost
         ) / self.num_substations
 
     def calc_mpt_cost(self):
-        """Computes transformer cost"""
+        """Computes HVAC main power transformer (MPT). MPT cost is 0 for HVDC.
+
+        Parameters
+        ----------
+        mpt_cost : int | float
+        """
+
+        _mpt_cost = self._design.get("mpt_cost", 2.87e6)
 
         self.num_mpt = self.num_cables
-        self.mpt_cost = self.num_mpt * self._design.get(
-            "mpt_cost_rate", 1750000
-        )
+
+        if "HVDC" in self.cable.cable_type:
+            self.mpt_cost = 0
+
+        else:
+            self.mpt_cost = self.num_mpt * _mpt_cost
+
         self.mpt_rating = (
             round((self._plant_capacity * 1.15 / self.num_mpt) / 10.0) * 10.0
         )
 
     def calc_shunt_reactor_cost(self):
-        """Computes shunt reactor cost"""
+        """Computes HVAC shunt reactor cost. Shunt reactor cost is 0 for HVDC.
+
+        Parameters
+        ----------
+        shunt_cost_rate : int | float
+        """
 
         touchdown = self.config["site"]["distance_to_landfall"]
+        shunt_cost_rate = self._design.get("shunt_cost_rate", 1e4)
 
-        if (
-            self.cable.cable_type == "HVDC-monopole"
-            or self.cable.cable_type == "HVDC-bipole"
-        ):
-            compensation = 0
+        _num_shunts = self.num_cables
+
+        if "HVDC" in self.cable.cable_type:
+            self.compensation = 0
         else:
             for _, cable in self.cables.items():
-                compensation = touchdown * cable.compensation_factor  # MW
+                self.compensation = touchdown * cable.compensation_factor  # MW
         self.shunt_reactor_cost = (
-            compensation
-            * self._design.get("shunt_cost_rate", 99000)
-            * self.num_cables
+            self.compensation * shunt_cost_rate * _num_shunts
         )
 
     def calc_switchgear_costs(self):
-        """Computes switchgear cost"""
+        """Computes HVAC switchgear cost. Switchgear cost is 0 for HVDC.
+
+        Parameters
+        ----------
+        switchgear_cost : int | float
+        """
 
-        if (
-            self.cable.cable_type == "HVDC-monopole"
-            or self.cable.cable_type == "HVDC-bipole"
-        ):
+        switchgear_cost = self._design.get("switchgear_cost", 4e6)
+
+        if "HVDC" in self.cable.cable_type:
             num_switchgear = 0
         else:
             num_switchgear = self.num_cables
-        self.switchgear_cost = num_switchgear * self._design.get(
-            "switchgear_cost", 134000
-        )
+
```

Comment:
This is an area where I'd traditionally make this a one-liner like `num_switchgear = 0 if "HVDC" in self.cable.cable_type else self.num_cables`.

Reply:
I like it.